katdal/datasources.py

```diff
@@ -97,15 +113,30 @@ class ChunkStoreVisFlagsWeights(VisFlagsWeights):
     """
     def __init__(self, store, base_name, chunk_info):
         self.store = store
-        da = {}
+        darray = {}
+        extra_flags = []
         for array, info in chunk_info.iteritems():
             array_name = store.join(base_name, array)
-            da[array] = store.get_dask_array(array_name, info['chunks'],
-                                             info['dtype'])
-        vis = da['correlator_data']
-        flags = da['flags']
+            chunk_args = (array_name, info['chunks'], info['dtype'])
+            darray[array] = store.get_dask_array(*chunk_args)
+            # Find all missing chunks in array and convert to 'data_lost' flags
+            has_array = store.has_dask_array(*chunk_args)
+            chunks_lost = da.map_blocks(_has_chunk_to_flags, has_array,
+                                        token='missing-chunks-' + array_name,
+                                        chunks=info['chunks'], dtype=np.uint8,
+                                        full_chunks=info['chunks'])
+            # If a flag chunk is missing but not the corresponding vis/weights,
+            # the data_lost bit will *not* be set (all flags cleared, actually)
```

Comment:
This was @sratcliffe's suggestion, the thinking being that egregiously bad data will eventually get flagged again somewhere down the line. There is also the L1 flags, essentially enhanced copies of the L0 flags, that will be folded in soon, thereby mitigating missing flag chunks.

The conclusion of our discussion is that I'll eventually fold in L1 and if both are missing, we'll flag it `data_lost` (since this will hopefully happen very infrequently).

Reply:
The latest thinking is: use L1 flags in preference to L0 flags if available, instead of trying to combine them somehow.